### PR TITLE
octoprint: 1.5.3 -> 1.6.1

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Updates OctoPrint to the latest version.

OctoPrints pins a lot of its dependencies to older versions, causing it to often fail to build when packages are updated in nixpkgs. Recently, the Pallets Projects (Flask, Werkzeug, Click, etc.) all released new major versions, which are not officially supported by OctoPrint. Most of the version restrictions appear to exist not for any specific reason, but just to avoid any potential breakages that might occur when upgrading to an new major version. A few other dependencies are pinned to keep support for Python 2, which is not relevant to us.

In this PR, I chose to be fairly aggressive about removing the version pinning and have not run into any problems as a result. All the unit tests pass, and I've printed with it without issue.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @abbradar @gebner @WhittlesJr @lovesegfault

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132164` at 125a29b5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>octoprint</li>
    <li>python38Packages.immutabledict</li>
    <li>python38Packages.zipstream-new</li>
    <li>python39Packages.immutabledict</li>
    <li>python39Packages.zipstream-new</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/immutabledict/default.nix:15:3:
```
   |
15 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/zipstream-new/default.nix:15:3:
```
   |
15 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/immutabledict/default.nix:16:0:
```
   |
16 |     description = "A fork of frozendict, an immutable wrapper around dictionaries.";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/zipstream-new/default.nix:16:0:
```
   |
16 |     description = "A zip archive generator";
   | ^
```
</li>
  </ul>
</details>

**Comment by lovesegfault** [comment]
A very welcome improvement!

File: pkgs/development/python-modules/immutabledict/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  pname = "immutabledict";
+  version = "2.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0ylnddj3x1zvak4fp69bk2c56k3zbfgrf2q5jk9kvlj61zrvhgv7";
+  };
+
+  # No tests in PyPI archive
+  doCheck = false;
```
Comment (lovesegfault) [inline pkgs/development/python-modules/immutabledict/default.nix:13]: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

File: pkgs/development/python-modules/zipstream-new/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pname = "zipstream-new";
+  version = "1.1.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0xhw24zhibmyypcdbmvql02jcf3npisb29lx71w1drcl3ccgwcdh";
+  };
+
+  # No tests in PyPI archive
+  doCheck = false;
```
Comment (lovesegfault) [inline pkgs/development/python-modules/zipstream-new/default.nix:13]: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

File: pkgs/development/python-modules/zipstream-new/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  meta = with lib; {
```
Comment (lovesegfault) [inline pkgs/development/python-modules/zipstream-new/default.nix:15]: Add yourself to the maintainers, if you don't mind.

File: pkgs/development/python-modules/immutabledict/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  meta = with lib; {
```
Comment (lovesegfault) [inline pkgs/development/python-modules/immutabledict/default.nix:15]: Add yourself to the maintainers, if you don't mind.

**Review by lovesegfault — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/immutabledict/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/immutabledict/default.nix:15]: We won't accept packages without maintainers.

File: pkgs/applications/misc/octoprint/default.nix [hunk 5]
```diff
@@ -113,13 +110,13 @@ let
...
               pname = "OctoPrint";
-              version = "1.5.3";
+              version = "1.6.1";
 
               src = fetchFromGitHub {
                 owner = "OctoPrint";
                 repo = "OctoPrint";
                 rev = version;
-                sha256 = "sha256-ZL/P/YIHynPmP8ssZZUKZDJscBsSsCq3UtOHrTVLpec=";
+                sha256 = "0mrh7z38wyjfliq2icl4pzb196n5k7r0yxd91h7jvyajc0zmfrvw";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/default.nix:119]:
```suggestion
                sha256 = "04rwm2s6x2c4b320ylqvx0cd5n8089xabzzxzigjmx873zvf9gfx";
```

**Comment by mweinelt** [comment]
I went that route (aggressively unpinning dependencies) but you really need the older flask version, otherwise blueprints error out, and that means you need to override all palletsprojects packages.

File: pkgs/applications/misc/octoprint/default.nix (hunk 5 above)
Comment (lopsided98) [reply pkgs/applications/misc/octoprint/default.nix]: Not quite sure what happened here, but I updated the hash.

**Comment by lopsided98** [comment]
I haven't run into that issue and am not particularly inclined to try to pin all those packages. Since the package is already broken, one possibility is to merge this PR and mark the package as broken. Users who acknowledge that it might not work can override the broken attribute and use it anyway.

**Comment by mweinelt** [comment]
I just rebased what I did ~2 weeks ago and opened a PR. PTAL.

**Comment by mweinelt** [comment]
Updated and fixed the build in #135302.

**Pull request closed without merging**